pkg/ingester/ingester.go

```diff
@@ -635,7 +648,7 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, req *mimirpb.WriteReques
 		// and out of order support is not enabled.
 		// TODO(jesus.vazquez) If we had too many old samples we might want to
 		// extend the fast path to fail early.
-		if i.cfg.BlocksStorageConfig.TSDB.OOOAllowance == 0 && minAppendTimeAvailable &&
+		if oooAllowance == 0 && minAppendTimeAvailable &&
```

Comment:
Shall we check for `oooAllowance <= 0` instead?

Reply:
Fixed in https://github.com/grafana/mimir/pull/2139/commits/6c3c72c28a55c2b80aaa62700e235acc94d68841